include/oneapi/tbb/detail/_flow_graph_impl.h

```diff
@@ -526,7 +526,8 @@ inline graph_task* prioritize_task(graph& g, graph_task& gt) {
     //! priority queue, and a new critical task is created to take and execute a work item with
     //! the highest known priority. The reference counting responsibility is transferred to
     //! the new task.
-    d1::task* critical_task = gt.my_allocator.new_object<priority_task_selector>(g.my_priority_queue, gt.my_allocator);
+    d1::small_object_allocator allocator;
+    d1::task* critical_task = allocator.new_object<priority_task_selector>(g.my_priority_queue, allocator);
```

Comment:
At the very least I would add a comment why it is necessary, explaining why graph_task's allocator cannot be used here. After all, its purpose is to cache allocations. BTW, if it cannot be used, do we need to have it in `graph_task` at all?

Reply:
Sure, I agree that the comment is required.
Regarding the necessity of the allocator inside of the `graph_task`, it is required not only for allocating the critical tasks, but also for deallocation of the `graph_task` itself after the execution (even in the case without critical tasks).

Reply:
The comment was added